### ISSUE TITLE
MinigetError: input stream: Status code: 404

### ISSUE THREAD

**Issue by Nico-ITA** [fent/node-ytdl-core#939]
MinigetError: input stream: Status code: 404
    at ClientRequest.<anonymous>
    at Object.onceWrapper (events.js:422:26)
    at ClientRequest.emit (events.js:315:20)
    at HTTPParser.parserOnIncomingClient (_http_client.js:641:27)
    at HTTPParser.parserOnHeadersComplete (_http_common.js:126:17)
    at TLSSocket.socketOnData (_http_client.js:509:22)
    at TLSSocket.emit (events.js:315:20)
    at addChunk (internal/streams/readable.js:309:12)
    at readableAddChunk (internal/streams/readable.js:284:9)
    at TLSSocket.Readable.push (internal/streams/readable.js:223:10) {
  statusCode: 404
}
I have this issue evry time i try to play  a yt song ( mp3 file work and soundcloud songs work too) . I have this issue today and I havn't touch the code ; only show this error in the console

**Comment by parthasarath1** [comment]
Yes, Even I am facing same issue

**Comment by a26364177** [comment]
this pr can fixed
https://github.com/fent/node-ytdl-core/pull/938/commits/d1f535712410eebd79a7468abd07d94f8c4ebe5a

**Comment by Nico-ITA** [comment]
Somone on the support discord server say this : "  404 due to youtube updates their site and ytdl-core haven't been updated yet ". I think is correct becouse every single song that I try to play throw this error

**Comment by Nico-ITA** [comment]
> 
> 
> this pr can fixed
> [d1f5357](https://github.com/fent/node-ytdl-core/commit/d1f535712410eebd79a7468abd07d94f8c4ebe5a)



> 
> 
> this pr can fixed
> [d1f5357](https://github.com/fent/node-ytdl-core/commit/d1f535712410eebd79a7468abd07d94f8c4ebe5a)
fixed omg XD

**Comment by Exhabition** [comment]
That fix has some side effects, `.lengthSeconds` is no longer available for me atleast

**Comment by 404invalid-user** [comment]
ah so its not just me I have been trouble shooting for hours

**Comment by nguyenHao5495** [comment]
Yes, Even I am facing same issue

**Comment by Samiul30** [comment]
> Yes, Even I am facing same issue

I am aslo facing this.

**Comment by Wolowit** [comment]
same

[fent/node-ytdl-core#923](https://github.com/fent/node-ytdl-core/issues/923)

**Comment by Nico-ITA** [comment]
> 
> 
> Yes, Even I am facing same issue

try this https://github.com/fent/node-ytdl-core/commit/d1f535712410eebd79a7468abd07d94f8c4ebe5a ot jus wait new update

**Comment by nguyenHao5495** [comment]
> > Yes, Even I am facing same issue
> 
> try this [d1f5357](https://github.com/fent/node-ytdl-core/commit/d1f535712410eebd79a7468abd07d94f8c4ebe5a) ot jus wait new update
I am using ytdl-core to download youtube videos. I don't know how to handle this error

**Comment by 404invalid-user** [comment]
> > > Yes, Even I am facing same issue
> > 
> > 
> > try this [d1f5357](https://github.com/fent/node-ytdl-core/commit/d1f535712410eebd79a7468abd07d94f8c4ebe5a) ot jus wait new update
> > I am using ytdl-core to download youtube videos. I don't know how to handle this error

I have been trying an on error doesn't work you can't even try catch

**Comment by JeHwanYoo** [comment]
I think I found the problem.

<img src="https://i.ibb.co/CbW9BkN/2021-06-22-9-14-01.png" alt="2021-06-22-9-14-01" border="0">

I tracked functions in the library, and when I took a log, the URL was strange...

plz check `function getVideoInfoPage()` in ytdl-core/lib/info.js

**Comment by JeHwanYoo** [comment]
I think I know what's wrong.

<img src="https://i.ibb.co/wsqFnqP/2021-06-22-9-20-47.png" alt="2021-06-22-9-20-47" border="0">

The URL "https://www.youtube.com/get_video_info" must work. But now responding with a 404.

need to check if the policy of the YouTube API has changed.

**Comment by shansekai** [comment]
`.lengthSeconds` undefined

**Comment by javalsai** [comment]
> That fix has some side effects, `.lengthSeconds` is no longer available for me at least

Also having trouble with `.publishDate`

**Comment by Nico-ITA** [comment]
> 
> 
> > That fix has some side effects, `.lengthSeconds` is no longer available for me at least
> 
> Also having trouble with `.publishDate` and `.related_videos` return an empty array
> Likes and dislikes are `null` and the channel has `undefined` subs

wait new update

**Comment by zaskker** [comment]
Tengo el mismo error, esperando nueva actualizacion.

**Comment by Dxngaa** [comment]
I have been researching and this error is due to YouTube having changed API. It is time to be patient.

**Comment by EPXOL** [comment]
This error can be fixed by not using getVideoInfoPage (set line 64 to //)
Then  you can bypass too many requests error by disabling getWatchJSONPage (set line 63 to //)
Then only site that can be datadownloaded from is WatchHTMLPage but it's corupted so we need to fix it go to line 254 and replace it with ```js
let jsonStr = body.split("ytInitialPlayerResponse = {")[1].split("}}};")[0] += "}}}";
```
All of edits are in node_modules/ytdl-core/info.json

Whole code here:

```js
const querystring = require('querystring');
const sax = require('sax');
const miniget = require('miniget');
const utils = require('./utils');
// Forces Node JS version of setTimeout for Electron based applications
const { setTimeout } = require('timers');
const formatUtils = require('./format-utils');
const urlUtils = require('./url-utils');
const extras = require('./info-extras');
const sig = require('./sig');
const Cache = require('./cache');


const BASE_URL = 'https://www.youtube.com/watch?v=';


// Cached for storing basic/full info.
exports.cache = new Cache();
exports.cookieCache = new Cache(1000 * 60 * 60 * 24);
exports.watchPageCache = new Cache();


// Special error class used to determine if an error is unrecoverable,
// as in, ytdl-core should not try again to fetch the video metadata.
// In this case, the video is usually unavailable in some way.
class UnrecoverableError extends Error {}


// List of URLs that show up in `notice_url` for age restricted videos.
const AGE_RESTRICTED_URLS = [
  'support.google.com/youtube/?p=age_restrictions',
  'youtube.com/t/community_guidelines',
];


/**
 * Gets info from a video without getting additional formats.
 *
 * @param {string} id
 * @param {Object} options
 * @returns {Promise<Object>}
*/
exports.getBasicInfo = async(id, options) => {
  const retryOptions = Object.assign({}, miniget.defaultOptions, options.requestOptions);
  options.requestOptions = Object.assign({}, options.requestOptions, {});
  options.requestOptions.headers = Object.assign({},
    {
      // eslint-disable-next-line max-len
      'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.101 Safari/537.36',
    }, options.requestOptions.headers);
  const validate = info => {
    let playErr = utils.playError(info.player_response, ['ERROR'], UnrecoverableError);
    let privateErr = privateVideoError(info.player_response);
    if (playErr || privateErr) {
      throw playErr || privateErr;
    }
    return info && info.player_response && (
      info.player_response.streamingData || isRental(info.player_response) || isNotYetBroadcasted(info.player_response)
    );
  };
  let info = await pipeline([id, options], validate, retryOptions, [
    getWatchHTMLPage,
    //getWatchJSONPage,
    //getVideoInfoPage,
  ]);

  Object.assign(info, {
    formats: parseFormats(info.player_response),
    related_videos: extras.getRelatedVideos(info),
  });

  // Add additional properties to info.
  const media = extras.getMedia(info);
  const additional = {
    author: extras.getAuthor(info),
    media,
    likes: extras.getLikes(info),
    dislikes: extras.getDislikes(info),
    age_restricted: !!(media && media.notice_url && AGE_RESTRICTED_URLS.some(url => media.notice_url.includes(url))),

    // Give the standard link to the video.
    video_url: BASE_URL + id,
    storyboards: extras.getStoryboards(info),
    chapters: extras.getChapters(info),
  };

  info.videoDetails = extras.cleanVideoDetails(Object.assign({},
    info.player_response && info.player_response.microformat &&
    info.player_response.microformat.playerMicroformatRenderer,
    info.player_response && info.player_response.videoDetails, additional), info);

  return info;
};

const privateVideoError = player_response => {
  let playability = player_response && player_response.playabilityStatus;
  if (playability && playability.status === 'LOGIN_REQUIRED' && playability.messages &&
    playability.messages.filter(m => /This is a private video/.test(m)).length) {
    return new UnrecoverableError(playability.reason || (playability.messages && playability.messages[0]));
  } else {
    return null;
  }
};


const isRental = player_response => {
  let playability = player_response.playabilityStatus;
  return playability && playability.status === 'UNPLAYABLE' &&
    playability.errorScreen && playability.errorScreen.playerLegacyDesktopYpcOfferRenderer;
};


const isNotYetBroadcasted = player_response => {
  let playability = player_response.playabilityStatus;
  return playability && playability.status === 'LIVE_STREAM_OFFLINE';
};


const getWatchHTMLURL = (id, options) => `${BASE_URL + id}&hl=${options.lang || 'en'}`;
const getWatchHTMLPageBody = (id, options) => {
  const url = getWatchHTMLURL(id, options);
  return exports.watchPageCache.getOrSet(url, () => utils.exposedMiniget(url, options).text());
};


const EMBED_URL = 'https://www.youtube.com/embed/';
const getEmbedPageBody = (id, options) => {
  const embedUrl = `${EMBED_URL + id}?hl=${options.lang || 'en'}`;
  return utils.exposedMiniget(embedUrl, options).text();
};


const getHTML5player = body => {
  let html5playerRes =
    /<script\s+src="([^"]+)"(?:\s+type="text\/javascript")?\s+name="player_ias\/base"\s*>|"jsUrl":"([^"]+)"/
      .exec(body);
  return html5playerRes ? html5playerRes[1] || html5playerRes[2] : null;
};


const getIdentityToken = (id, options, key, throwIfNotFound) =>
  exports.cookieCache.getOrSet(key, async() => {
    let page = await getWatchHTMLPageBody(id, options);
    let match = page.match(/(["'])ID_TOKEN\1[:,]\s?"([^"]+)"/);
    if (!match && throwIfNotFound) {
      throw new UnrecoverableError('Cookie header used in request, but unable to find YouTube identity token');
    }
    return match && match[2];
  });


/**
 * Goes through each endpoint in the pipeline, retrying on failure if the error is recoverable.
 * If unable to succeed with one endpoint, moves onto the next one.
 *
 * @param {Array.<Object>} args
 * @param {Function} validate
 * @param {Object} retryOptions
 * @param {Array.<Function>} endpoints
 * @returns {[Object, Object, Object]}
 */
const pipeline = async(args, validate, retryOptions, endpoints) => {
  let info;
  for (let func of endpoints) {
    try {
      const newInfo = await retryFunc(func, args.concat([info]), retryOptions);
      if (newInfo.player_response) {
        newInfo.player_response.videoDetails = assign(
          info && info.player_response && info.player_response.videoDetails,
          newInfo.player_response.videoDetails);
        newInfo.player_response = assign(info && info.player_response, newInfo.player_response);
      }
      info = assign(info, newInfo);
      if (validate(info, false)) {
        break;
      }
    } catch (err) {
      if (err instanceof UnrecoverableError || func === endpoints[endpoints.length - 1]) {
        throw err;
      }
      // Unable to find video metadata... so try next endpoint.
    }
  }
  return info;
};


/**
 * Like Object.assign(), but ignores `null` and `undefined` from `source`.
 *
 * @param {Object} target
 * @param {Object} source
 * @returns {Object}
 */
const assign = (target, source) => {
  if (!target || !source) { return target || source; }
  for (let [key, value] of Object.entries(source)) {
    if (value !== null && value !== undefined) {
      target[key] = value;
    }
  }
  return target;
};


/**
 * Given a function, calls it with `args` until it's successful,
 * or until it encounters an unrecoverable error.
 * Currently, any error from miniget is considered unrecoverable. Errors such as
 * too many redirects, invalid URL, status code 404, status code 502.
 *
 * @param {Function} func
 * @param {Array.<Object>} args
 * @param {Object} options
 * @param {number} options.maxRetries
 * @param {Object} options.backoff
 * @param {number} options.backoff.inc
 */
const retryFunc = async(func, args, options) => {
  let currentTry = 0, result;
  while (currentTry <= options.maxRetries) {
    try {
      result = await func(...args);
      break;
    } catch (err) {
      if (err instanceof UnrecoverableError ||
        (err instanceof miniget.MinigetError && err.statusCode < 500) || currentTry >= options.maxRetries) {
        throw err;
      }
      let wait = Math.min(++currentTry * options.backoff.inc, options.backoff.max);
      await new Promise(resolve => setTimeout(resolve, wait));
    }
  }
  return result;
};


const jsonClosingChars = /^[)\]}'\s]+/;
const parseJSON = (source, varName, json) => {
  if (!json || typeof json === 'object') {
    return json;
  } else {
    try {
      json = json.replace(jsonClosingChars, '');
      return JSON.parse(json);
    } catch (err) {
      throw Error(`Error parsing ${varName} in ${source}: ${err.message}`);
    }
  }
};


const findJSON = (source, varName, body, left, right, prependJSON) => {
  let jsonStr = body.split("ytInitialPlayerResponse = {")[1].split("}}};")[0] += "}}}";
  if (!jsonStr) {
    throw Error(`Could not find ${varName} in ${source}`);
  }
  return parseJSON(source, varName, utils.cutAfterJSON(`${prependJSON}${jsonStr}`));
};


const findPlayerResponse = (source, info) => {
  const player_response = info && (
    (info.args && info.args.player_response) ||
    info.player_response || info.playerResponse || info.embedded_player_response);
  return parseJSON(source, 'player_response', player_response);
};


const getWatchJSONURL = (id, options) => `${getWatchHTMLURL(id, options)}&pbj=1`;
const getWatchJSONPage = async(id, options) => {
  const reqOptions = Object.assign({ headers: {} }, options.requestOptions);
  let cookie = reqOptions.headers.Cookie || reqOptions.headers.cookie;
  reqOptions.headers = Object.assign({
    'x-youtube-client-name': '1',
    'x-youtube-client-version': '2.20201203.06.00',
    'x-youtube-identity-token': exports.cookieCache.get(cookie || 'browser') || '',
  }, reqOptions.headers);

  const setIdentityToken = async(key, throwIfNotFound) => {
    if (reqOptions.headers['x-youtube-identity-token']) { return; }
    reqOptions.headers['x-youtube-identity-token'] = await getIdentityToken(id, options, key, throwIfNotFound);
  };

  if (cookie) {
    await setIdentityToken(cookie, true);
  }

  const jsonUrl = getWatchJSONURL(id, options);
  const body = await utils.exposedMiniget(jsonUrl, options, reqOptions).text();
  let parsedBody = parseJSON('watch.json', 'body', body);
  if (parsedBody.reload === 'now') {
    await setIdentityToken('browser', false);
  }
  if (parsedBody.reload === 'now' || !Array.isArray(parsedBody)) {
    throw Error('Unable to retrieve video metadata in watch.json');
  }
  let info = parsedBody.reduce((part, curr) => Object.assign(curr, part), {});
  info.player_response = findPlayerResponse('watch.json', info);
  info.html5player = info.player && info.player.assets && info.player.assets.js;

  return info;
};


const getWatchHTMLPage = async(id, options) => {
  let body = await getWatchHTMLPageBody(id, options);
  let info = { page: 'watch' };
  try {
    info.player_response = findJSON('watch.html', 'player_response',
      body, /\bytInitialPlayerResponse\s*=\s*\{/i, '\n', '{');
  } catch (err) {
    let args = findJSON('watch.html', 'player_response', body, /\bytplayer\.config\s*=\s*{/, '</script>', '{');
    info.player_response = findPlayerResponse('watch.html', args);
  }
  info.response = findJSON('watch.html', 'response', body, /\bytInitialData("\])?\s*=\s*\{/i, '\n', '{');
  info.html5player = getHTML5player(body);
  return info;
};


const INFO_HOST = 'www.youtube.com';
const INFO_PATH = '/get_video_info';
const VIDEO_EURL = 'https://youtube.googleapis.com/v/';
const getVideoInfoPage = async(id, options) => {
  const url = new URL(`https://${INFO_HOST}${INFO_PATH}`);
  url.searchParams.set('video_id', id);
  url.searchParams.set('eurl', VIDEO_EURL + id);
  url.searchParams.set('ps', 'default');
  url.searchParams.set('gl', 'US');
  url.searchParams.set('hl', options.lang || 'en');
  url.searchParams.set('html5', '1');
  const body = await utils.exposedMiniget(url.toString(), options).text();
  let info = querystring.parse(body);
  info.player_response = findPlayerResponse('get_video_info', info);
  return info;
};


/**
 * @param {Object} player_response
 * @returns {Array.<Object>}
 */
const parseFormats = player_response => {
  let formats = [];
  if (player_response && player_response.streamingData) {
    formats = formats
      .concat(player_response.streamingData.formats || [])
      .concat(player_response.streamingData.adaptiveFormats || []);
  }
  return formats;
};


/**
 * Gets info from a video additional formats and deciphered URLs.
 *
 * @param {string} id
 * @param {Object} options
 * @returns {Promise<Object>}
 */
exports.getInfo = async(id, options) => {
  let info = await exports.getBasicInfo(id, options);
  const hasManifest =
    info.player_response && info.player_response.streamingData && (
      info.player_response.streamingData.dashManifestUrl ||
      info.player_response.streamingData.hlsManifestUrl
    );
  let funcs = [];
  if (info.formats.length) {
    info.html5player = info.html5player ||
      getHTML5player(await getWatchHTMLPageBody(id, options)) || getHTML5player(await getEmbedPageBody(id, options));
    if (!info.html5player) {
      throw Error('Unable to find html5player file');
    }
    const html5player = new URL(info.html5player, BASE_URL).toString();
    funcs.push(sig.decipherFormats(info.formats, html5player, options));
  }
  if (hasManifest && info.player_response.streamingData.dashManifestUrl) {
    let url = info.player_response.streamingData.dashManifestUrl;
    funcs.push(getDashManifest(url, options));
  }
  if (hasManifest && info.player_response.streamingData.hlsManifestUrl) {
    let url = info.player_response.streamingData.hlsManifestUrl;
    funcs.push(getM3U8(url, options));
  }

  let results = await Promise.all(funcs);
  info.formats = Object.values(Object.assign({}, ...results));
  info.formats = info.formats.map(formatUtils.addFormatMeta);
  info.formats.sort(formatUtils.sortFormats);
  info.full = true;
  return info;
};


/**
 * Gets additional DASH formats.
 *
 * @param {string} url
 * @param {Object} options
 * @returns {Promise<Array.<Object>>}
 */
const getDashManifest = (url, options) => new Promise((resolve, reject) => {
  let formats = {};
  const parser = sax.parser(false);
  parser.onerror = reject;
  let adaptationSet;
  parser.onopentag = node => {
    if (node.name === 'ADAPTATIONSET') {
      adaptationSet = node.attributes;
    } else if (node.name === 'REPRESENTATION') {
      const itag = parseInt(node.attributes.ID);
      if (!isNaN(itag)) {
        formats[url] = Object.assign({
          itag, url,
          bitrate: parseInt(node.attributes.BANDWIDTH),
          mimeType: `${adaptationSet.MIMETYPE}; codecs="${node.attributes.CODECS}"`,
        }, node.attributes.HEIGHT ? {
          width: parseInt(node.attributes.WIDTH),
          height: parseInt(node.attributes.HEIGHT),
          fps: parseInt(node.attributes.FRAMERATE),
        } : {
          audioSampleRate: node.attributes.AUDIOSAMPLINGRATE,
        });
      }
    }
  };
  parser.onend = () => { resolve(formats); };
  const req = utils.exposedMiniget(new URL(url, BASE_URL).toString(), options);
  req.setEncoding('utf8');
  req.on('error', reject);
  req.on('data', chunk => { parser.write(chunk); });
  req.on('end', parser.close.bind(parser));
});


/**
 * Gets additional formats.
 *
 * @param {string} url
 * @param {Object} options
 * @returns {Promise<Array.<Object>>}
 */
const getM3U8 = async(url, options) => {
  url = new URL(url, BASE_URL);
  const body = await utils.exposedMiniget(url.toString(), options).text();
  let formats = {};
  body
    .split('\n')
    .filter(line => /^https?:\/\//.test(line))
    .forEach(line => {
      const itag = parseInt(line.match(/\/itag\/(\d+)\//)[1]);
      formats[line] = { itag, url: line };
    });
  return formats;
};


// Cache get info functions.
// In case a user wants to get a video's info before downloading.
for (let funcName of ['getBasicInfo', 'getInfo']) {
  /**
   * @param {string} link
   * @param {Object} options
   * @returns {Promise<Object>}
   */
  const func = exports[funcName];
  exports[funcName] = async(link, options = {}) => {
    utils.checkForUpdates();
    let id = await urlUtils.getVideoID(link);
    const key = [funcName, id, options.lang].join('-');
    return exports.cache.getOrSet(key, () => func(id, options));
  };
}


// Export a few helpers.
exports.validateID = urlUtils.validateID;
exports.validateURL = urlUtils.validateURL;
exports.getURLVideoID = urlUtils.getURLVideoID;
exports.getVideoID = urlUtils.getVideoID;
```

**Comment by JerryFunHouse** [comment]
Tried the fix, now I dont have any sound

**Comment by EPXOL** [comment]
This has nothing to do with video or audio downloading it fixes data (info) downloading and that was the problem. And I am using it in my bot and it works.

**Comment by CrypticMrDark** [comment]
well, he's right. follow Epxol

**Comment by EdgarNunezTrejo** [comment]
I fixed it adding this two lines:
`url.searchParams.set('c', 'TVHTML5');
  url.searchParams.set('cver', '7.20190319');`

As @a26364177 refered, then I put:

 `try{
 var info = await ytdl.getInfo(url)
}catch(e){
 console.log('error catch',e)
}` 

And it worked again.

**Comment by mozzi-berilinky** [comment]
I tried the fix. And some videos play and others don't with this error.
![제목 없음](https://user-images.githubusercontent.com/67187351/123016814-5cfe6400-d406-11eb-934c-f0a926e943c6.png)
This error has never happened before. What is the problem

**Comment by EPXOL** [comment]
As I sayed it has nothing to do with downloading of video/audio. I was getting this error before too and wasn't even using the fix idk why it hapens.

**Comment by ROSTHUMANGAMES** [comment]
i cnt fix it pls dm me in disocrd pls sir or mam on Vaibhav dev#9662 😭

**Comment by EPXOL** [comment]
We fixed it on discord then

**Comment by lbarcl** [comment]
Is there an error on new versions?

**Comment by mtripg6666tdr** [comment]
While waiting an update, we can call [`youtube-dl`](https://github.com/ytdl-org/youtube-dl) binary by `exec` (child_process) with `--skip-download` and `--print-json` args to get video info as json.
(The video info also contains temporary video urls in each format.)

**Comment by cchen028** [comment]
For those that aren't able to directly modify the node_modules folder using the solution provided by @EPXOL  
(ex: deploying to Google App Engine)
A workaround is to fork then push your change and reference the fork repository in your `package.json`

ex: `"hack-node-ytdl-core": "git+https://github.com/<username>/node-ytdl-core.git"`

**Comment by MoonLGH** [comment]
> For those that aren't able to directly modify the node_modules folder using the solution provided by @EPXOL  
> (ex: deploying to Google App Engine)
> A workaround is to fork then push your change and reference the fork repository in your `package.json`
> 
> ex: `"hack-node-ytdl-core": "git+https://github.com/<username>/node-ytdl-core.git"`

Did someone has a pr for that ?
I want to just fork it lol

**Comment by cchen028** [comment]
> > For those that aren't able to directly modify the node_modules folder using the solution provided by @EPXOL
> > (ex: deploying to Google App Engine)
> > A workaround is to fork then push your change and reference the fork repository in your `package.json`
> > ex: `"hack-node-ytdl-core": "git+https://github.com/<username>/node-ytdl-core.git"`
> 
> Did someone has a pr for that ?
> I want to just fork it lol

Fork mine. Replace <username> with cchen028

**Comment by PasinduDushan** [comment]
> > > For those that aren't able to directly modify the node_modules folder using the solution provided by @EPXOL
> > > (ex: deploying to Google App Engine)
> > > A workaround is to fork then push your change and reference the fork repository in your `package.json`
> > > ex: `"hack-node-ytdl-core": "git+https://github.com/<username>/node-ytdl-core.git"`
> > 
> > 
> > Did someone has a pr for that ?
> > I want to just fork it lol
> 
> Replace `username` with **cchen028**

It's still giving me 404 error. 
```"ytdl-core": "git+https://github.com/cchen028/node-ytdl-core.git"```

**Comment by BIOLOGY-SCIENCE** [comment]
So im not the only one with this problem

**Comment by PasinduDushan** [comment]
> So im not the only one with this problem

Yeah. I think we should wait for an official release

**Comment by D3SOX** [comment]
@Exhabition @shansekai I've pushed a fix for the `lengthSeconds` being undefined issue. Can you confirm that it works now?

@javalsai `related_videos` seems to be set for me. Can you send a video ID with which it doesn't work?

I can however reproduce the issue with `publishDate`.
`player_response.microformat` seems to undefined which is critical.
`likes`, `dislikes` and `subscriber_count` also seems to be undefined now. I'm looking into it right now.

**Comment by EvasiveXkiller** [comment]
@EPXOL Thank you at least its working up until a certain point

**Comment by a26364177** [comment]
@D3SOX
**related_videos**  always empty array in almost video.
eg. Yhyp-_hX2s

**Comment by NEXGENGAMER** [comment]
why is the error coming and is there any fix

**Comment by D3SOX** [comment]
> Yhyp-_hX2s

this is not a valid video id

**Comment by a26364177** [comment]
> > Yhyp-_hX2s
> 
> this is not a valid video id

_Yhyp-_hX2s

> > Yhyp-_hX2s
> 
> this is not a valid video id

_Yhyp-_hX2s

**Comment by NEXGENGAMER** [comment]
why i a getting this error ? MinigetError: Status code: 404

**Comment by D3SOX** [comment]
@a26364177 Hm, I see. Now it's empty for me too even on the video which previously worked. I don't know if we can do anything about this.

**Comment by ben186** [comment]
Apparently it has something to do with age-gating, can anyone confirm this? https://github.com/ytdl-org/youtube-dl/issues/29372

**Comment by nukeop** [comment]
Nuclear's users have been reporting this issue as well (the YT update is probably being rolled out progressively or A/B testing is involved so not everyone is affected).

**Comment by Lazy-C0der** [comment]
> This has nothing to do with video or audio downloading it fixes data (info) downloading and that was the problem. And I am using it in my bot and it works.

he's right

**Comment by Lazy-C0der** [comment]
> why i a getting this error ? MinigetError: Status code: 404

epxol has already given code above 
but if you still have doubt then try this 

https://youtu.be/TYMsAYAx5Z0

**Comment by CrypticMrDark** [comment]
i'm not saying you are wrong
but there are many who get trouble doing this so instead of finding 2 and
doing copy paste
it's easy to delete all and then paste.

i hope you understand
thanks for mailing :)

**Comment by deereyeskart** [comment]
Song and videos not downloading,, something went error occur.. Please fix the error.. Please new update suddenly.. Thank you

**Comment by RaghuMudem** [comment]
When is the official release for version 4.8.3 with the fix for #940  ??

**Comment by ryzyx** [comment]
Has this issue been fixed yet?

**Comment by jbelford** [comment]
@EPXOL I believe the root of this issue here is that the use of regex to get a substring seems to be very flaky - particularly since we have a regex that tries to find the _end_ of the object definition which regex is not well suited for. And the object representation changing will break it each time.

If we just get the point where this variable is created we could probably just do something like below to get to the beginning of the object definition.
```js
let jsonStr = body.split(/(var|let|const)\s+ytInitialPlayerResponse\s*=\s*\/i)[1];
```
Then what we want to do is iterate through the object definition like is done in `cutAfterJSON` (we may be able to straight up call this as is without any changes). We don't need to know when the end of this object definition is in the string - if we reach the end of the string we error out.

TLDR; we can likely fix bug by not using `utils.between(body, left, right)` instead just get substring of whole string after `left`

**Comment by D3SOX** [comment]
@jbelford #946 already fixes this part of the problem in a better way

**Comment by sunii-g** [comment]
good, it works great

**Comment by TimeForANinja** [comment]
as noted in #938 i hope the 404s are fixed
various videoDetails are broken, but i wanna keep the discussion for that in https://github.com/fent/node-ytdl-core/issues/958
thanks for everyone that helped

**Comment by QuazArxx** [comment]
I'm still getting 404 errors. I just started attempting to use ytdl-core. I tried the fix above by EPXOL and while the error went away, there is still no sound coming from links having that issue.

**Comment by Nico-ITA** [comment]
> I'm still getting 404 errors. I just started attempting to use ytdl-core. I tried the fix above by EPXOL and while the error went away, there is still no sound coming from links having that issue.

Hi Its come again for me too becouse I try to play restricted video age

**Comment by QuazArxx** [comment]
> > I'm still getting 404 errors. I just started attempting to use ytdl-core. I tried the fix above by EPXOL and while the error went away, there is still no sound coming from links having that issue.
> 
> Hi Its come again for me too becouse I try to play restricted video age

I didn't notice until you said that but yeah the issues seem to be with restricted videos now. All the other videos work fine.

**Comment by Nico-ITA** [comment]
> > > I'm still getting 404 errors. I just started attempting to use ytdl-core. I tried the fix above by EPXOL and while the error went away, there is still no sound coming from links having that issue.
> > 
> > 
> > Hi Its come again for me too becouse I try to play restricted video age
> 
> I didn't notice until you said that but yeah the issues seem to be with restricted videos now. All the other videos work fine.

Yes same situation

**Comment by nukeop** [comment]
A pox upon youtube, their executives, and theirs for these shenaningans

**Comment by TimeForANinja** [comment]
might be related but i'd still prefer a new, own issue for that then 😉

**Comment by sanikava** [comment]
@Nico-ITA Hi now we are getting new error 401 
lol

**Comment by Nico-ITA** [comment]
> @Nico-ITA Hi now we are getting new error 401
> lol

Hey bro ! Im only getting 403 error xD never 401

**Comment by nokisenpai** [comment]
Here I get 410 error :) 

`D:\Perso\prog\Kinoa\node_modules\miniget\dist\index.js:210
                let err = new Miniget.MinigetError(Status code: ${res.statusCode}, res.statusCode);
                          ^

MinigetError: Status code: 410
    at ClientRequest.<anonymous> (D:\Perso\prog\Kinoa\node_modules\miniget\dist\index.js:210:27)
    at Object.onceWrapper (node:events:510:26)
    at ClientRequest.emit (node:events:390:28)
    at HTTPParser.parserOnIncomingClient (node:_http_client:623:27)
    at HTTPParser.parserOnHeadersComplete (node:_http_common:128:17)
    at TLSSocket.socketOnData (node:_http_client:487:22)
    at TLSSocket.emit (node:events:390:28)
    at addChunk (node:internal/streams/readable:324:12)
    at readableAddChunk (node:internal/streams/readable:297:9)
    at TLSSocket.Readable.push (node:internal/streams/readable:234:10)
Emitted 'error' event on PassThrough instance at:
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  statusCode: 410
}`

**Comment by sanikava** [comment]
> 

it means your access forbidden because of high api use 😓 just stop using heroku if you using otherwise wait for a long time 🤣
